tests/validators/regex_validator_test.py

```diff
@@ -161,6 +161,86 @@ def test_invalid_regex_pattern():
             RegexValidator('[')
         assert str(exception_info.value) == 'unterminated character set at position 0'
 
+    # Test RegexValidator with output template
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'regex_pattern, output_template, input_data, expected_output', [
+            # Empty template returns the original string
+            (r'.*', r'', 'some input', 'some input'),
```

Comment:
I think if the output_template is explicitly specified as an empty string (as opposed to `None`), the validator should also return a constant empty string. At least that's what I would expect, since a constant non-empty string will also be returned as it is.

So, if output_template is None, return the original string, otherwise use the output_template even if it's empty.

Reply:
This particular testcase documents the surprising behavior of `match.expand`. Honestly I was surprised, too. I guessed at least I could put it in the tests for completeness/documentation.

We can of course not follow `match.expand`'s peculiar behavior. I'll try and address it.